sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -1224,6 +1224,15 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
       }
     }
   }
+
+  test("SPARK-29000: arithmetic computation overflow when don't allow decimal precision loss ") {
+    val allow = sql("set spark.sql.decimalOperations.allowPrecisionLoss")
+    val originValue = allow.collect().apply(0).apply(1)
+    sql("set spark.sql.decimalOperations.allowPrecisionLoss=false")
+    val df = sql("select case when 1=2 then 1 else 100.000000000000000000000000 end * 1")
+    QueryTest.checkAnswer(df, Array(Row(100)))
+    sql(s"set spark.sql.decimalOperations.allowPrecisionLoss=$originValue")
+  }
```

Comment:
Why did you add tests in package `hive`? I think this issue is not only for hive (I didn't carefully check yet)

Reply:
I have moved it into SQLQuerySuite in  package `sql`. Thanks.

Reply:
I think we need tests in `TypeCoercionSuite`, too.

Reply:
I have added relative tests in TypeCoericionSuite.